packages/core/chrome/core-chrome-browser-internal/src/ui/project/app_menu.tsx

```diff
@@ -17,26 +17,43 @@ interface AppMenuBarProps {
 }
 export const AppMenuBar = ({ headerActionMenuMounter }: AppMenuBarProps) => {
   const { euiTheme } = useEuiTheme();
+  const [height, setHeight] = React.useState<number | null>(null);
+
+  React.useLayoutEffect(() => {
+    if (height === null) return;
+    const kbnAppWrapper = document.querySelector('.kbnAppWrapper') as HTMLElement;
+    if (!kbnAppWrapper) return; // should never happen
+
+    kbnAppWrapper.style.setProperty('--kbnProjectHeaderAppActionMenuHeight', `${height}px`);
+    return () => {
+      kbnAppWrapper.style.removeProperty('--kbnProjectHeaderAppActionMenuHeight');
+    };
+  }, [height]);
+
   return (
-    <div
-      className="header__actionMenu"
-      data-test-subj="kibanaProjectHeaderActionMenu"
-      css={css`
-        z-index: ${euiTheme.levels.header};
-        background: ${euiTheme.colors.emptyShade};
-        border-bottom: ${euiTheme.border.thin};
-        display: flex;
-        justify-content: end;
-        align-items: center;
-        padding: ${euiTheme.size.s};
-        margin-bottom: -${euiTheme.border.width.thin};
-        /* fixates the elements position in the viewport, removes the element from the flow of the page */
-        position: sticky;
-        /* position below the primary fixed EuiHeader in the viewport */
-        top: var(--euiFixedHeadersOffset, 0);
-      `}
-    >
-      <HeaderActionMenu mounter={headerActionMenuMounter} />
-    </div>
+    <EuiResizeObserver onResize={(d) => setHeight(d.height)}>
+      {(resizeRef) => (
+        <div
+          className="header__actionMenu"
+          data-test-subj="kibanaProjectHeaderActionMenu"
+          ref={resizeRef}
+          css={css`
+            z-index: ${euiTheme.levels.header};
+            background: ${euiTheme.colors.emptyShade};
+            border-bottom: ${height ? euiTheme.border.thin : 'none'};
+            justify-content: end;
+            align-items: center;
+            padding: ${height ? euiTheme.size.s : 0};
```

Comment:
Not needed for https://github.com/elastic/kibana/issues/167908, but just an improvement. We remove the padding and border to make the container visibly hidden in case there is no real content inside